src/Uno.UI.Runtime.Skia.Wpf/Rendering/OpenGLWpfRenderer.cs

```diff
@@ -43,7 +43,19 @@ public OpenGLWpfRenderer(IWpfXamlRootHost host)
 	public bool TryInitialize()
 	{
 		// Get the window from the wpf control
-		var hwnd = new WindowInteropHelper(Window.GetWindow(_hostControl)).Handle;
+		var window = Window.GetWindow(_hostControl);
+		if (window is null)
+		{
+			throw new InvalidOperationException("The host control is not associated with any Window");
+		}
+
+		var windowInteropHelper = new WindowInteropHelper(window);
+		var hwnd = windowInteropHelper.Handle;
+		if (hwnd == IntPtr.Zero)
```

Comment:
What the `windowInteropHelper.EnsureHandle` do is just like your code:

```csharp
    public sealed class WindowInteropHelper
    {
        public IntPtr Handle
        {
            get
            {
                return CriticalHandle;
            }
        }

        /// <summary>
        /// Create the hwnd of the Window if the hwnd is not created yet.
        /// </summary>
        public IntPtr EnsureHandle()
        {

            if (CriticalHandle == IntPtr.Zero)
            {
                _window.CreateSourceWindow(false /*create hwnd during show*/);
            }

            return CriticalHandle;
        }
  }
```

Reply:
Adjusted, now also throwing in case the HWND is still zero (which should now never be the case)